### PR TITLE
Role unit tests

### PR DESCRIPTION
**Pull request by code-incubus**
Signed-off-by: code-incubus nemanja.milutinovic@comtrade.com

Added unit test scenarios for Role Service.

**Related Issue**
_None_

**Description of the solution adopted**
Test scenarios that are added in this PR are checking functionalities regarding Roles. All scenarios are added to _RoleServiceUnitTests.feature_. Scenarios are testing if Roles are correctly created, edited or deleted. 

**Screenshots**
/

**Any side note on the changes made**
/

### CODE REVIEW

File: service/security/test/src/test/resources/features/RoleServiceUnitTests.feature [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+     Given I prepare a role creator with name "roleName" and description "roleDescription"
+     When I create a new role entity from the existing creator
+     Then I delete the role with name "roleName"
+     But I expect the exception "KapuaEntityNotFoundException" with the text "*"
+     And I update the role description to "roleDescription123"
+     And An exception was thrown
+
+   Scenario: Changing role name to null
+     Create a role with a valid name (e.g "roleName") and with a valid description (e.g "roleDescription").
+     After that tr to update role name to "null".
```
Comment (LeoNerdoG) [inline service/security/test/src/test/resources/features/RoleServiceUnitTests.feature]: Change description to: " After that try to update role name to "null"."

File: service/security/test/src/test/resources/features/RoleServiceUnitTests.feature [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+     Given I prepare a role creator with name "roleName" and description "roleDescription"
+     When I create a new role entity from the existing creator
+     But I expect the exception "KapuaIllegalNullArgumentException" with the text "An illegal null value was provided for the argument role.name."
+     Then I update the role name to ""
+     And An exception was thrown
+
+   Scenario: Changing role name to contain special character
+     Create a role with name e.g "roleName" and with description e.g "roleDescription",
+     then try to update role name with special characters e.g "roleName$%^".
```
Comment (LeoNerdoG) [inline service/security/test/src/test/resources/features/RoleServiceUnitTests.feature]: Change description line to: then try to update role name with special characters( e.g "roleName%")."

File: service/security/test-steps/src/main/java/org/eclipse/kapua/service/authorization/steps/AuthorizationServiceSteps.java [hunk 3]
```diff
@@ -2534,4 +2535,162 @@ public void iCreateTheAccessInfoEntities() throws Exception {
...
+                stepData.remove("Role");
+                Role role = roleService.create(roleCreator);
+                stepData.put("Role", role);
+            } catch (KapuaException ex) {
+                verifyException(ex);
+            }
+        }
+    }
+
+    @Then("^I update the role name with special characters$")
```
Comment (LeoNerdoG) [inline service/security/test-steps/src/main/java/org/eclipse/kapua/service/authorization/steps/AuthorizationServiceSteps.java]: Please change step name to: " update the role name with special characters ""#$%&'(‹›€ı–......."

File: service/security/test-steps/src/main/java/org/eclipse/kapua/service/authorization/steps/AuthorizationServiceSteps.java [hunk 4]
```diff
@@ -2534,4 +2535,162 @@ public void iCreateTheAccessInfoEntities() throws Exception {
...
+    public void iUpdateTheRoleNameWithSpecialCharacters() throws Exception {
```
Comment (LeoNerdoG) [inline service/security/test-steps/src/main/java/org/eclipse/kapua/service/authorization/steps/AuthorizationServiceSteps.java]: **Please change implementation to:** 

```
public void iUpdateTheRoleNameWithSpecialCharacters(String invalidSymbols) throws Exception {
        RoleCreator roleCreator = roleFactory.newCreator(SYS_SCOPE_ID);
            for (int i = 0; i < invalidSymbols.length(); i++) {
                String roleName = "roleName" + invalidSymbols.charAt(i);
                 roleCreator.setName("roleName" + i);

            try {
                primeException();
                stepData.remove("Role");
                Role role = roleService.create(roleCreator);
                role.setName("roleName" + invalidSymbols.charAt(i));
                roleService.update(role);
                stepData.put("Role", role);
            } catch (KapuaException ex) {
                verifyException(ex);
            }
        }
    }
```

File: service/security/test/src/test/resources/features/RoleServiceUnitTests.feature [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+     Kapua should return an errors.
+
+     But I expect the exception "KapuaIllegalArgumentException" with the text "An illegal value was provided for the argument role.name"
+     Then I update the role name with special characters
```
Comment (LeoNerdoG) [inline service/security/test/src/test/resources/features/RoleServiceUnitTests.feature]: **Please change step name to:** 
Then I update the role name with special characters "!"#$%&'()=⁄@‹›€ı–......."

Please also change implementation!!!! (See comment above)

**Review by LeoNerdoG — CHANGES_REQUESTED** [review]
Hej @code-incubus,
I have reviewed this PR and there are some changes to be made. 
Plase fix this as soon as you can. 
@Coduz after the changes are made and all the tests pass and codeCove is OK, please merge this.

**Comment by code-incubus** [comment]
Done. @LeoNerdoG

**Review by Coduz — CHANGES_REQUESTED** [review]
Small thing.

Could you please remove `()` when declaring the `@Inject` annotation? 
Also, since the init of the `RoleServiceImpl` has changes, can you please check that the runtime is not affected by that? If you didn't do it already. Running a console and CRUDing a Role should be enough.

Thanks,

- Alberto

**Comment by code-incubus** [comment]
Hi @Coduz, @LeoNerdoG 
Checked. Everything works fine and all the changes are made.

**Comment by LeoNerdoG** [comment]
Hey @Coduz,
I have checked this PR and from QA stanpoint everything is OK. 
All the tests have passed, code coverage has increased, no CQ is needed, so I guess we can merge this. 

Leo

**Review by Coduz — APPROVED** [review]

**Review by LeoNerdoG — APPROVED** [review]

**Pull request merged**